edx_analytics_transformers/django/models.py

```diff
@@ -0,0 +1,165 @@
+"""
+Models for filtering of events
+"""
+import logging
+
+from django.db import models
+from model_utils.models import TimeStampedModel
+from jsonfield.fields import JSONField
+
+from edx_django_utils.cache import TieredCache, get_cache_key
+
+
+logger = logging.getLogger(__name__)
+
+
+ROUTER_CACHE_NAMESPACE = 'edx_analytics.request_router'
+
+
+def get_value_from_dotted_path(dict_obj, dotted_key):
+    """
+    Map the dotted key to nested keys for dict and return the matching value.
+
+    For example:
+        'key_a.key_b.key_c' will look for the folowing value:
+
+        {
+            'key_a': {
+                'key_b': {
+                    'key_c': 'final value'
+                }
+            }
+        }
+
+    Arguments:
+        dict_obj (dict)  :    dictionary for which the value is required
+        dotted_key (str) :    dotted key string for the dict
+
+    Returns:
+        ANY :                 depends upon the value in dict
+    """
+    nested_keys = dotted_key.split('.')
+    result = dict_obj
+    try:
+        for key in nested_keys:
+            result = result[key]
+    except KeyError:
+        return None
+    return result
+
+
+class RouterConfiguration(TimeStampedModel):
```

Comment:
Is this model a timestamped model so that we can keep the old history of configurations that used to be setup?  If so, why not just have a separate history table connected to the base model for this?

Reply:
Added `simple_history.models.HistoricalRecords` for history records.